drafts/developer-guide/pages/data-access-layer/index.adoc

```diff
@@ -0,0 +1,97 @@
+include::ROOT:partial$variables.adoc[]
+
+= Data Access Layer (DAL)
```

Comment:
Connections and modules, in the context of DAL, can report errors in a _special_ way.
Instead of showing an error message in the toast, it provides a list of **issues** in a separate UI section.

Module issues are shown under the `issues` tab on the module edit screen
![image](https://user-images.githubusercontent.com/12545711/192240014-5d0ff0ee-8caa-4de0-a03c-3edfd22c35f0.png)

Connection issues are shown at the bottom next to where you provide DB connection parameters
![image](https://user-images.githubusercontent.com/12545711/192240292-af859e1f-afdb-4a2d-983e-8815ba97e49d.png)


Issues are used to allow modules/connections to be misconfigured while allowing the system to continue working  (for example, if you restart the server with a misconfigured connection, it will still start up). It does, however, prevent specific operations such as creating records for a broken module configuration.